english_tasks/irregular_verbs.py

```diff
@@ -74,52 +102,81 @@ def display_statistic(list1, list2, list3):
     print(f'Fixed verbs:   ', end="")
     for i in list3:
         print(i, end=",")
+    print(f'\nYour scores is {scores}%')
+
+
+def get_args():
+    a = ArgumentParser()
+    a.add_argument('-f', '--filename', dest='verbs_filename', type=str, required=True)
+    return a.parse_args()
+
+
+def generate_attempts(nb_attempt, verb, form_nb):
+    for attempt in range(1, nb_attempt + 1):
+        print(f'Attempt №{attempt}')
+        next_answer = input(f'Enter V{form_nb} again: ')
+        if compare_verb(next_answer, verb):
+            print('Correct!')
+            RESULT_DICT['Fixed Verbs'].append(verb)
+            return True
+    RESULT_DICT['Wrong Verbs'].append(verb)
+    return False
 
 
 def main():
-    list_of_rows = rows_list('irregular_verbs')
-    while len(list_of_rows) != 0:
-        current_verb = random_verb(list_of_rows)
-        print(current_verb[0])
-        user_answer = input('Enter here: ')
-        if user_answer == "q":
-            break
-        user_verbs = user_answer.split()
-
-        # check every word from user's input(you should def a function later)
-        result = verbs_comparing(user_verbs, current_verb[1:])
-
-        if not result:
-            wrong_verbs.append(current_verb[0])
-            ask_hint = input('Do you want to get a hint?(yes/no): ')
-            if ask_hint.lower() == 'yes':
-                print(current_verb)
-            next_try = input('Do you want to do it again? (yes/no): ')
-            while next_try != 'no':
-                user_answer = input('Enter here: ')
-                user_verbs = user_answer.split()
-                result = verbs_comparing(user_verbs, current_verb[1:])
-                if result:
-                    fixed_verbs.append(current_verb[0])
-                    wrong_verbs.remove(current_verb[0])
-                    break
-                else:
-                    next_try = input('Do you want to do it again? (yes/no): ')
-
-        else:
-            right_verbs.append(current_verb[0])
-        list_of_rows.remove(current_verb)
-    ask_stat = input('Do you want to see a statistic?(yes/no): ')
-    if ask_stat == 'yes':
-        display_statistic(right_verbs, wrong_verbs, fixed_verbs)
+    arguments = get_args()
 
+    # read the special file to store verbs into a list
+    verbs_list = rows_list(arguments.verbs_filename)
+    greeting()
+    name = input('Enter your name: ')
+    nb_verb = int(input('Enter the number of verbs: '))
 
-if __name__ == '__main__':
-    main()
+    if not 0 < nb_verb <= len(verbs_list):
+        print('Error! You entered a wrong value (max number is 74)')
+        sys.exit(1)
```

Comment:
too complex, do this:
- get overall quantity of irregular verbs from a file save them into list
- use random.shuffle() to shuffle the list
- iterate over a list of irregular verbs for the user to practice